src/transaction/lock_manager.c

```diff
@@ -7703,6 +7703,11 @@ void
 lock_force_thread_timeout_lock (THREAD_ENTRY * thrd)
 {
 #if defined (SERVER_MODE)
+  if (!logtb_get_check_interrupt (thrd))
+    {
+      // don't interrupt
+      return;
+    }
```

Comment:
How about reverting this code and retry in case of LK_NOTGRANTED_DUE_TIMEOUT in case of lock_object call in xbtree_load_online_index ?

Reply:
@eseokoh what do you agree with alex's proposal? it is less risky than current fix.

Reply:
Yes, indeed.